cloud/endagaweb/tasks.py

```diff
@@ -213,9 +214,23 @@ def update_credit(self, imsi, update_id):
         if request.status_code >= 200 and request.status_code < 300:
             print "update_credit SUCCESS. id=%s, imsi=%s, amount=%s. (%d)" % (
                 update_id, imsi, update.amount, request.status_code)
-            update.delete()
-            bts.mark_active()
-            bts.save()
+            with transaction.atomic():
+                expiry_date = update.valid_through
+                update.subscriber.state = 'active'
+                # Get subscriber's 1st number from some admin number.
+                num = Number.objects.filter(
+                    subscriber__imsi=update.subscriber.imsi)[0:1].get()
+                if num.valid_through is None:
```

Comment:
1) This field doesn't exist it seems in `Number`. Probably better to put it under `Subscriber` since `Numbers` can be recycled
2) None validation should be done by the model to ensure its never None